src/rompy/run/slurm.py

```diff
@@ -0,0 +1,298 @@
+                logger.error("sbatch command not found. SLURM may not be installed or in PATH.")
+                return None
+
+            # Check if SLURM controller is responsive
+            result = subprocess.run(
+                ["squeue", "--help"],
+                capture_output=True,
+                text=True,
+                timeout=10  # Don't wait too long
+            )
+            if result.returncode != 0:
+                logger.error("SLURM controller is not responsive. squeue command failed.")
+                return None
+
+            # Submit the job using sbatch
+            result = subprocess.run(
+                ["sbatch", job_script],
+                capture_output=True,
+                text=True,
+                check=True
+            )
+
+            # Extract job ID from sbatch output (format: "Submitted batch job <ID>")
+            output = result.stdout.strip()
+            if "Submitted batch job" in output:
+                job_id = output.split()[-1]
+                logger.info(f"Submitted SLURM job with ID: {job_id}")
+                return job_id
+            else:
+                logger.error(f"Unexpected sbatch output format: {output}")
+                return None
+
+        except subprocess.TimeoutExpired:
+            logger.error("SLURM controller check timed out. SLURM may not be properly configured.")
+            return None
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Failed to submit SLURM job: {e.stderr}")
+            return None
+        except Exception as e:
+            logger.error(f"Error submitting SLURM job: {e}")
+            return None
+        finally:
+            # Clean up the temporary job script
+            try:
+                os.remove(job_script)
+                logger.debug(f"Cleaned up temporary job script: {job_script}")
+            except OSError:
+                logger.warning(f"Could not remove temporary job script: {job_script}")
+
+    def _wait_for_completion(self, job_id: str, config: "SlurmConfig") -> bool:
+        """Wait for job completion.
+
+        Args:
+            job_id: SLURM job ID to monitor
+            config: SlurmConfig with timeout parameters
+
+        Returns:
+            True if job completed successfully, False otherwise
+        """
+        logger.info(f"Waiting for SLURM job {job_id} to complete...")
+
+        # Terminal states that indicate job completion (successful or failed)
+        terminal_states = {'CD', 'CA', 'F', 'TO', 'NF', 'OOM', 'BF', 'DL', 'PR'}
+
+        # Start time for timeout check
+        start_time = time.time()
+
+        while True:
+            # Check if we've exceeded the timeout
+            elapsed_time = time.time() - start_time
+            if elapsed_time > config.timeout:
+                logger.error(f"Timeout waiting for job {job_id} after {config.timeout} seconds")
+                
+                # Try to cancel the job
+                try:
```

Comment:
Just rethinking this, this is probably not a good idea. Python should not be actively trying to kill the job after the timeout, we should leave that to slurm